lib/format_parser.rb

```diff
@@ -62,7 +62,9 @@ def self.register_parser(callable_parser, formats:, natures:, priority: LEAST_PR
         @parsers_per_format[provided_format] << callable_parser
       end
       @parser_priorities ||= {}
+      @parser_registration_order ||= []
```

Comment:
I might be off, but couldn't we reuse `@parsers` by making it an array? It feels redundant to have two structures holding the exact same items, and arrays also provide set methods (wouldn't count performance here, since there're very few registered parsers to begin with).

Reply:
Good point!

I changed to be an array, but I'm adding a new element only if it's not there, to act as an Set:
```ruby
@parsers << callable_parser unless @parsers.include?(callable_parser)
```

I think this way will prevent us to introduce a regression.

Is it ok for you?

Reply:
Sounds reasonable 👍  I'd only check whether there are _really_ no other bits that expect `@parsers` to be a Set.

Reply:
from what I saw, it's ok using an array!